packages/core/src/runtime/utils.ts

```diff
@@ -23,14 +24,18 @@ export function removeBase(url: string): string {
   return rawRemoveBase(url, siteData.base);
 }
 
-export function isEqualPath(a: string, b: string) {
-  return withBase(normalizeHref(a)) === withBase(normalizeHref(b));
+export function isEqualPath(a: string, b: string, cleanUrls: boolean) {
+  return withBase(normalizeHref(a, cleanUrls)) === withBase(normalizeHref(b, cleanUrls));
+}
+
+export function useNormalizeHrefInRuntime(a: string){
+  const cleanUrls = !!usePageData()?.siteData?.route?.cleanUrls;
```

Comment:
It's better to use `Boolean()` instead of `!!` to avoid ambiguity and some of optional chain is not neccessary, such as:

```ts
const cleanUrls = Boolean(usePageData().siteData.route?.cleanUrls);
```

Reply:
Fixed it.